File: SevenNightClassPCS/oop-java/tictactoe1/MrRandom.java
import java.util.Random;

public class MrRandom {
	
	private Board brd;
	private Random rnd = new Random();

	public MrRandom(Board brd, char token) {
		this.brd = brd;		
	}
	
	public int getMove() {		
		while(true) {
			int move = rnd.nextInt(9);
			if(brd.getCell(move)==' ') {
				return move;
			}
		}		
	}

}

File: Topics/01_Encapsulation/01_05_TicTacToe_example/python_source/mr_random.py
# We don't need "random" in the main anymore. MrRandom needs it ... nobody
# else needs to worry about it

import random

class MrRandom:
    
    def __init__(self, token, board):        
        self._board = board
     
    def get_move(self):   
        while True:
            computer_move = random.randint(0, 8)
            if self._board.get_cell(computer_move) == ' ':
                return computer_move